Add support for labelNames and labelValues

vCra opened the pull request:
Fixes the labelValues and labelNames part of https://github.com/thanos-community/thanos-promql-connector/issues/5

Does not add support for Series

fpetkovski reviewed (approved):
Awesome. Did you get a chance to test this manually?

pedro-stanaka commented on .gitignore:
```diff
@@ -19,3 +19,6 @@
 
 # Go workspace file
 go.work
+
+# IntelliJ files
+.idea
```
This should be on your personal gitignore. `git config --global core.excludesfile ~/.gitignore_global`

vCra replied:
Done ✅

vCra committed:
Add support for labelNames and labelValues

Add implementation of store server to enable labelNames and labelValues

Undo ignoring .idea folder

pedro-stanaka reviewed (approved):
lg

sarthaktyagi-505 commented:
Hi Guys,

I am trying to extend this connector to also use Series endpoint, Now I have read previous comments and Series is a heavy call but I was wondering, that we might also extend a capability of aggregating the data from various sources (which is doesnt happen right now if I am not wrong). Am I correct to assume that if we have both store and query servers running as part of this, the query in thanos distributed mode will be able to use either of them depending on if the query is split or not? 

We just tried to use thanos-sidecar for this purpose but since sidecar is only storeAPI every call is a remote_read which makes querying through it really slow. What do you guys think?

fpetkovski commented:
Hi @sarthaktyagi-505, the Series call should not be needed for aggregating data between two query nodes. The proxy will distribute the query to both nodes and aggregate data centrally. Perhaps something  is not properly configured in your setup if you are unable to do that.
